diff -Nur caribou-0.4.20/configure caribou-0.4.20n/configure
--- caribou-0.4.20/configure	2016-02-16 07:15:27.000000000 +0800
+++ caribou-0.4.20n/configure	2016-07-07 10:46:40.732570466 +0800
@@ -14044,7 +14044,7 @@
 $as_echo_n "checking for intltool >= 0.35.0... " >&6; }
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INTLTOOL_APPLIED_VERSION found" >&5
 $as_echo "$INTLTOOL_APPLIED_VERSION found" >&6; }
-    test "$INTLTOOL_APPLIED_VERSION_AS_INT" -ge "$INTLTOOL_REQUIRED_VERSION_AS_INT" ||
+    test "$INTLTOOL_APPLIED_VERSION_AS_INT" -le "$INTLTOOL_REQUIRED_VERSION_AS_INT" ||
 	as_fn_error $? "Your intltool is too old.  You need intltool 0.35.0 or later." "$LINENO" 5
 fi
 
