diff -Nur epiphany-3.14.2/embed/web-extension/Makefile.am epiphany-3.14.2n/embed/web-extension/Makefile.am
--- epiphany-3.14.2/embed/web-extension/Makefile.am	2014-08-29 16:48:35.000000000 -0800
+++ epiphany-3.14.2n/embed/web-extension/Makefile.am	2014-11-14 19:03:27.707339732 -0800
@@ -29,7 +29,9 @@
 	$(top_srcdir)/lib/ephy-uri-helpers.c \
 	$(top_srcdir)/lib/ephy-uri-helpers.h \
 	$(top_srcdir)/lib/ephy-web-dom-utils.c \
-	$(top_srcdir)/lib/ephy-web-dom-utils.h
+	$(top_srcdir)/lib/ephy-web-dom-utils.h \
+	$(top_srcdir)/lib/ephy-profile-utils.c \
+	$(top_srcdir)/lib/ephy-profile-utils.h 
 
 libephywebextension_la_CPPFLAGS = \
 	-I$(top_srcdir)/embed \
@@ -39,6 +41,7 @@
 libephywebextension_la_CFLAGS = \
 	-DSHARE_DIR=\"$(pkgdatadir)\" \
 	-DTOP_SRC_DATADIR=\"$(top_srcdir)/data\" \
+	-DABS_TOP_BUILD_DIR=\"$(abs_top_builddir)\" \
 	$(AM_CFLAGS) \
 	$(WEB_EXTENSION_CFLAGS)
 
